message/message.go

```diff
@@ -0,0 +1,105 @@
+package message
+
+import "strings"
+
+// Message represents DNS messages
+type Message struct {
+	header  *header
+	queries []*query
+	Answers []*Record
+}
+
+type header struct {
+	ID      uint16
+	QR      byte
```

Comment:
If this is a byte for outside users (i.e, for anyone who is interpreting and using this information, should we really leak information about how to extract bits from a byte all throughout the project?